app/controllers/application_controller.rb

```diff
@@ -6,13 +6,32 @@ class ApplicationController < ActionController::Base
 
   def current_user
     return unless user_id.present?
-    @current_user ||= User.find_or_create_by(webauth: user_id)
+    @current_user ||= begin
+      user = User.find_or_create_by(webauth: user_id)
+      user.ldap_group_string = request_ldap if request_ldap
+      user
+    end
   end
   helper_method :current_user
 
   private
 
   def user_id
-    ENV['REMOTE_USER']
+    request.env['REMOTE_USER'] || ENV['REMOTE_USER']
+  end
+
+  def request_ldap
+    @request_ldap ||= begin
```

Comment:
Are you sure this construct works? At the very least, a `case` statement might be more clear.